sawmill-core/src/main/java/io/logz/sawmill/processors/Base64DecodeProcessor.java

```diff
@@ -0,0 +1,84 @@
+package io.logz.sawmill.processors;
+
+import io.logz.sawmill.Doc;
+import io.logz.sawmill.ProcessResult;
+import io.logz.sawmill.Processor;
+import io.logz.sawmill.annotations.ProcessorProvider;
+import io.logz.sawmill.exceptions.ProcessorConfigurationException;
+import io.logz.sawmill.utilities.JsonUtils;
+
+import java.util.Base64;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+
+@ProcessorProvider(type = "base64Decode", factory = Base64DecodeProcessor.Factory.class)
+public class Base64DecodeProcessor implements Processor {
+
+    private final Set<String> fields;
+    private final boolean allowMissingFields;
+
+    public Base64DecodeProcessor(Set<String> fields, boolean allowMissingFields) {
+        this.fields = requireNonNull(fields);
+        this.allowMissingFields = allowMissingFields;
+    }
+
+    @Override
+    public ProcessResult process(Doc doc) {
+        Set<String> missingFields = listMissingFields(doc);
+        if(!validateMissingFields(missingFields))
+            return ProcessResult.failure("some or all fields are missing from doc");
+
+        fields.stream()
+                .filter(field -> !missingFields.contains(field))
+                .forEach(field -> {
+                    String value = doc.getField(field);
+                    doc.addField(field, new String(Base64.getDecoder().decode(value)));
+                });
+        return ProcessResult.success();
+    }
+
+    private boolean validateMissingFields(Set<String> missingFields) {
```

Comment:
Would be a bit better to call it more `boolean` related. `areFieldsValid()` or something. WDYT?